adapters/ddf-adapter/src/main/java/com/connexta/replication/adapters/ddf/ResultIterable.java

```diff
@@ -179,9 +197,19 @@ private boolean isDistinctResult(@Nullable Metadata result) {
       return result != null && (result.getId() == null || !foundIds.contains(result.getId()));
     }
 
-    private void copyQueryRequestAndQuery(QueryRequest request, int index) {
+    private void nextRequestOrSignalFinished() {
```

Comment:
:question: Are we concerned by the recursion introduced between `fetchNextResults()` and `nextRequestOrSignalFinished()`? Should the logic be changed to avoid that?

Reply:
I asked @clockard about that and we seemed to agree that It shouldn't be something we need to worry about. It should only be us using this class and I would hope none of us would put a few thousand query requests into the result iterable.